code/design_pattern/src/factory/Roulette.java

```diff
@@ -0,0 +1,36 @@
+package factory;
+
+import factory.gifts.*;
+
+import java.util.Scanner;
+
+public class Roulette {
+    public void run() throws InterruptedException {
+        while (true) {
+            Scanner input = new Scanner(System.in);
+
+            System.out.println("Let's see what the russian roulette will give you");
+
+            System.out.println(generateGift(GiftType.randomGift()).message());
+
+            Thread.sleep(3000);
+        }
+    }
+
+    public Gift generateGift(GiftType gift) {
+        switch (gift) {
+            case Booze:
+                return new Booze();
+            case Car:
+                return new Car();
+            case Nothing:
+                return new Nothing();
+            case Toy:
+                return new Toy();
+            case Error:
```

Comment:
Why not remove the `Nothing` and `Error` cases?

Reply:
Because you could get Nothing as a gift ( it's a feature, part of the logic ) , while Error is thrown in case no String is matched in in GiftType's fromString.

Reply:
No I meant to remove the case since the default case handles both. (This may sacrifice readability, so the decision is up to you).